[KYUUBI 3269] [DOCS] Doc for JDBC authentication provider (#3275)
apache/incubator-kyuubi: docs/security/jdbc.md

<!--
 - Licensed to the Apache Software Foundation (ASF) under one or more
 - contributor license agreements.  See the NOTICE file distributed with
 - this work for additional information regarding copyright ownership.
 - The ASF licenses this file to You under the Apache License, Version 2.0
 - (the "License"); you may not use this file except in compliance with
 - the License.  You may obtain a copy of the License at
 -
 -   http://www.apache.org/licenses/LICENSE-2.0
 -
 - Unless required by applicable law or agreed to in writing, software
 - distributed under the License is distributed on an "AS IS" BASIS,
 - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 - See the License for the specific language governing permissions and
 - limitations under the License.
 -->

> # Configure Kyuubi to use JDBC Authentication

Reviewer: Use

> Kyuubi supports authentication via JDBC query. A query is prepared with username/password value and sent to database configured in JDBC URL. The authentication passes if result set returns at least one row.

Reviewer: ```suggestion
Kyuubi supports authentication via JDBC query. A query is prepared with username/password value and sent to the database configured in JDBC URL. The authentication passes if the result set is not empty.
```

> The query sql must start with `SELECT`. Placeholders are supported for sql template:

Reviewer: The SQL statement must start with the `SELECT` clause. Placeholders are supported and listed below for substitution

 - the License.  You may obtain a copy of the License at
 -
 -   http://www.apache.org/licenses/LICENSE-2.0
 -
 - Unless required by applicable law or agreed to in writing, software
 - distributed under the License is distributed on an "AS IS" BASIS,
 - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 - See the License for the specific language governing permissions and
 - limitations under the License.
 -->

# Configure Kyuubi to use JDBC Authentication

Kyuubi supports authentication via JDBC query with JDBC. A query is prepared with username/password value and sent to database configured in JDBC URL. The authentication passes if result set returns at least one row.

The query sql must start with `SELECT`. Placeholders are supported for sql template:
- `${username}`
- `${password}`

For example, `SELECT 1 FROM auth_db.auth_table WHERE user=${username} AND 
passwd=MD5(CONCAT(salt,${password}))` will be prepared as: `SELECT 1 FROM auth_db.auth_table WHERE user=? AND passwd=MD5(CONCAT(salt,?))` with value replacement of `username` and `password` in string type.

## Enable JDBC Authentication 

To enable the jdbc authentication method, we need to

> - put the JDBC driver jar package to ``$KYUUBI_HOME/jars`` directory to make it visible for

Reviewer 2: UPPER CASE?

Author: sure

Author: alright. lowered.

  the classpath of the kyuubi server.
- Configure the following properties to ``$KYUUBI_HOME/conf/kyuubi-defaults.conf``
  on each node where kyuubi server is installed.

## Configure the authentication properties

> Configure the following properties to `$KYUUBI_HOME/conf/kyuubi-defaults.conf` on each node where kyuubi server is installed.

Reviewer 2: nit: empty line

Author: ok
